sdk/core/Azure.Core.Experimental/src/MutableJsonElement.WriteTo.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Text.Json;
+
+namespace Azure.Core.Json
+{
+    public partial struct MutableJsonElement
+    {
+        internal void WriteTo(Utf8JsonWriter writer)
+        {
+            WriteElement(_path, _highWaterMark, _element, writer);
+        }
+
+        private void WriteElement(string path, int highWaterMark, JsonElement element, Utf8JsonWriter writer)
+        {
+            if (Changes.TryGetChange(path, highWaterMark, out MutableJsonChange change))
+            {
+                element = change.AsJsonElement();
```

Comment:
We can potentially improve performance here by switching on the type of change.Value and writing the value directly to the writer if it is a primitive, e.g. doesn't have descendants.  This will save us a round-trip through JsonSerializer.Serialize() and JsonDocument.Parse().  Noting this optimization for a future PR.